luigi/task.py

```diff
@@ -647,12 +661,15 @@ def input(self):
 
     def deps(self):
         """
-        Internal method used by the scheduler.
```

Comment:
Wait why?

Can we start with a smaller PR just renaming `deps` to `_deps` or something? Or maybe even inline it at places it is used (if apporpirate).

Reply:
The main purpose of `deps` seems to be overridden in `Wrapped` class to handle sub-task dependencies.

Reply:
@Tarrasch Raised #2685 to address this. Can you take a look?